core/transactions/attempt_context_impl.cxx

```diff
@@ -635,8 +643,9 @@ attempt_context_impl::remove_staged_insert(const core::document_id& id, VoidCall
     };
     debug("removing staged insert {}", id);
 
-    if (auto err = hooks_.before_remove_staged_insert(this, id.key()); err) {
-        error_handler(*err, "before_remove_staged_insert hook returned error");
+    auto err = hooks_.before_remove_staged_insert(this, id.key());
+    if (err) {
+        error_handler(*err, "before_remove_staged_insert hook returned error", std::move(cb));
```

Comment:
why the scope for `err` was expanded here? the old way seems to be okay

Reply:
Oh -- ya it was -- actually I liked it better that way.   I'll change it back!